core/domain/summary_services.py

```diff
@@ -201,7 +201,7 @@ def get_displayable_exp_summary_dicts(exploration_summaries):
     valid (i.e., none of them are None).
     """
     exploration_ids = [
-        exploration_summary.id
+        exploration_summary.id if exploration_summary else None
```

Comment:
Hmm, the docstring says that this function assumes all the exploration summary domain objects passed in are valid. So, the check should really be happening before you pass the exploration_summaries inside.